### PR TITLE
fixes bug 1241286 - Rate-limit for authenticated users too

### PR DESCRIPTION
**Pull request by peterbe**
cc @lonnen

### CODE REVIEW

File: webapp-django/crashstats/api/tests/test_views.py [hunk 1]
```diff
@@ -1634,29 +1634,45 @@ def mocked_get(url, params, **options):
...
-        # double to avoid https://bugzilla.mozilla.org/show_bug.cgi?id=1148470
-        for __ in range(current_limit * 2):
-            response = self.client.get(url)
-        eq_(response.status_code, 429)
+        with self.settings(
+            API_RATE_LIMIT='3/m',
+            API_RATE_LIMIT_AUTHENTICATED='6/m'
+        ):
+            # the rate limit is currently X/min so it's easy to hit the limit
+            current_limit = int(re.findall('\d+', settings.API_RATE_LIMIT)[0])
```
Comment (adngdb) [inline webapp-django/crashstats/api/tests/test_views.py]: Why do you bother doing a regex when you hard-coded the value 3 lines above?

Comment (adngdb) [inline webapp-django/crashstats/api/tests/test_views.py]: Why do you bother doing a regex when you hard-coded the value 3 lines above?

File: webapp-django/crashstats/api/tests/test_views.py [hunk 2]
```diff
@@ -1634,29 +1634,45 @@ def mocked_get(url, params, **options):
...
+            # double to avoid
+            # https://bugzilla.mozilla.org/show_bug.cgi?id=1148470
+            for __ in range(current_limit * 2):
+                response = self.client.get(url)
+            eq_(response.status_code, 429)
+
+            # but it'll work if you use a different X-Forwarded-For IP
+            response = self.client.get(url, HTTP_X_FORWARDED_FOR='11.11.11.11')
```
Comment (adngdb) [inline webapp-django/crashstats/api/tests/test_views.py]: This is unclear to me. What does it do?

File: webapp-django/crashstats/api/tests/test_views.py [hunk 3]
```diff
@@ -1634,29 +1634,45 @@ def mocked_get(url, params, **options):
...
-        response = self.client.get(url, HTTP_AUTH_TOKEN=token.key)
-        eq_(response.status_code, 200)
+            for __ in range(current_limit):
+                response = self.client.get(url)
+            eq_(response.status_code, 200)
 
-        for __ in range(current_limit):
-            response = self.client.get(url)
-        eq_(response.status_code, 200)
+            # but even being signed in as a limit
```
Comment (adngdb) [inline webapp-django/crashstats/api/tests/test_views.py]: ``as a limit`` => ``has a limit``.

**Comment by adngdb** [comment]
r+ with nits.

**Comment by peterbe** [comment]
I addressed the nits and rebased and squashed.

**Pull request merged**